instance/github.py

```diff
@@ -149,26 +140,48 @@ class PR:
     """
     Representation of a GitHub Pull Request
     """
-    # pylint: disable=too-many-arguments
-    def __init__(self, number, source_fork_name, target_fork_name, branch_name, title, username, body=''):
+
+    def __init__(self, target_fork_name, number):
         self.number = number
-        self.fork_name = source_fork_name
         self.repo_name = target_fork_name
-        self.branch_name = branch_name
-        self.title = title
-        self.username = username
-        self.body = body
+        self.update_data()
+
+    def update_data(self):
+        self._data = get_pr_by_number(target_fork_name, number)
+        self._extra_settings = get_settings_from_pr_body(self.body)
+        self._extra_settings_dict = yaml.load(self.extra_settings) or {}
```

Comment:
@antoviaque It's just a difference in assessing risks, which might be related to the different backgrounds we are coming from.  I've seen many problems arising from performance issues like this in the past, so I tend to fix them right away if it's easy to do and I deem it safe, since doing it later might be more work than getting it right now.  I'm also aware of problems with data getting out of sync, but I think this is just one class of problem to keep in mind, not the only one and maybe not even the most important one.

My background is in high-performance computing and at Google; performance was paramount for both of these.  You are probably right that performance simply doesn't matter here, and I don't think we need to discuss this any further.  We should focus on solving actual problems instead. :)

I'll prepare a stripped-down version of this PR anyway with a few clean-ups, but I won't include caching the settings.